### ISSUE TITLE
Bugs with Saving product information properly

### ISSUE THREAD

**Issue by paulstroot** [ocupop/comission-application#145]
Find and identify bugs with saving data between form pages, and retaining saved info between pages. 

Will be referencing a some bullet points from other issues to point to this one.

**Comment by ZiggyAtl** [comment]
@paulstroot I've got some more examples for you on this one.

I imagined the museum wanted to go back and redo an artwork, but there is no way to delete an artwork once you've started. So I tried deleting some of the information and saving changes. The button went to grey and it looked like it had saved the changes, but when I made more changes I couldn’t save them as the Save button was now inactive. 

I then started all over again adding a new artwork. I saved some changes as I went along but by the time I got the the Review page there were a few bits needing saving, so I hit Save Changes, it went grey and remained spinning around with Saving on the button.

**Comment by paulstroot** [comment]
Thanks Bill. Still working out some of the bugs with saving. Occasionally it throws an error, but the screen doesn't alert the user that an error happened and it gets stuck on the "submitting" state. Thanks for the heads up.

**Comment by ZiggyAtl** [comment]
@paulstroot  a bit more info. I entered a new piece of art, and initially it let me save things without problem. Then when I entered the pricing tiers I had a problem. The button went to grey, with 'Saving' showing, and the timer spinning. And it stayed like that. However I was able to click on the Next box to get to the Review page, and all the pricing tier I had entered was showing, even though the Save button still said Saving.

One other piece of info. I made the artwork 89 columns x 39 rows, so almost 3500 tiles. Maybe it struggled handling the volume?

**Comment by paulstroot** [comment]
Hi @ZiggyAtl . I just made an update to the museum app that should fix a lot of the issues you were encountering with the submit hanging up staying on the spinner. 

You were correct that 89 columns and 39 rows was a lot of volume for it to handle. Now, when updating rows or columns, you will see a "re-fractionalize art" button to click, and that should re-process the DABs. Let me know if that works for you.

**Comment by ZiggyAtl** [comment]
@paulstroot I tried creating a new Add Art. Everything went smoothly up to the Review page (and I saved a few times as I was going along). The tier pricing section worked fine, and I had 89 x 39 with 3 pricing tiers. It still hung though when I hit save on the last page (Review). I went in to Shopify and the piece I had just entered appeared there. However when I went to the Manage Art screen in the app it was not there. And when I went back in to the Add Art screen it had lost everything I had entered and saved up to that point.

**Comment by ZiggyAtl** [comment]
@paulstroot anther example. This time it wouldn't allow me to select the Description box. I filled everything else in and moved past that screen. Decided to keep number of tiles low and only 2 price tiers. It then hung as I tried to save it on the Price Tier page. So frustrating that there doesn't seem to be any pattern to this.

![Screen Shot 2022-11-08 at 4 08 21 PM](https://user-images.githubusercontent.com/112654923/200676459-c8a294e1-4ddd-4a33-93af-8eb256a642bf.png)

**Comment by paulstroot** [comment]
With some of my latest updates I have not been running into this problem. I fixed a number of issues that may have resulted in the "spinning submit button of doom". And if there is an error, there should hopefully be a message that pops up letting you know. I'm going to mark this as "ready for review" now, but Let's keep this open for a few days just in case something pops up.

If you run into any issues with saving information, please continue to give as many details as you can about what step you were at, and if there are any steps I can take to try to recreate. Thanks!

**Comment by ZiggyAtl** [comment]
@paulstroot we're definitely getting there. I uploaded a piece and was able to save changes right up to the final review stage, and then submit for CoMissioning. I found one issue though;

Once you have set the price tier information and saved changes, if you ever go back to the Fractionalize Art page it will delete all that information, even if you do nothing on the Fractionalize page. This is the only page that happens on, and you can go back to any of the others without it losing that info.

**Comment by paulstroot** [comment]
Thanks Ziggy. I wasn't able to recreate the issue, but I did notice that if you do not click the "refractionalize image" button after changing the rows and columns, then leave and come back to that step, it reverts back to the last calculated rows and columns. I just made an update so that those numbers are preserved even if you leave and come back too the step. Just make sure to click the "refractionalize image" button before saving.
